components/Buttons/MarkWatchedButton.vue

```diff
@@ -0,0 +1,72 @@
+<template>
+  <v-btn v-if="canMarkWatched" icon dark @click.stop.prevent="togglePlayed">
+    <v-icon>
+      {{ isPlayed ? 'mdi-check-all' : 'mdi-check' }}
+    </v-icon>
+  </v-btn>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { mapActions } from 'vuex';
+import { BaseItemDto } from '@jellyfin/client-axios';
+
+export default Vue.extend({
+  props: {
+    item: {
+      type: Object as () => BaseItemDto,
+      required: true
+    }
+  },
+  data() {
+    return {
+      isPlayed: false
+    };
+  },
+  computed: {
+    canMarkWatched(): boolean {
+      if (
+        ['Series', 'Season', 'BoxSet', 'AudioPodcast', 'AudioBook'].includes(
+          this.item.Type || ''
+        )
+      ) {
+        return true;
+      }
+
+      if (this.item.MediaType === 'Video' && this.item.Type !== 'TvChannel') {
+        return true;
+      }
+
+      return false;
+    }
+  },
+  created() {
+    this.isPlayed = this.item.UserData?.Played || false;
+  },
+  methods: {
+    ...mapActions('snackbar', ['pushSnackbarMessage']),
+    async togglePlayed(): Promise<void> {
+      try {
+        if (this.isPlayed) {
+          await this.$api.playState.markUnplayedItem({
+            userId: this.$auth.user.Id,
+            itemId: this.item.Id || ''
+          });
+          this.isPlayed = false;
```

Comment:
I'd set `this.isPlayed` before the await call, and revert it in the catch. It'll be smoother for the user

Reply:
I disagree, this is the same as the like button, and it makes more sense to set it after a success


I wonder what @ferferga thinks?

Reply:
I've never checked in that case, and I agree we need to be consistent. But then I'll say the same for the like button ;p

To me, a user should always have a direct reaction to something. If not, he'll click multiple times. So either we do this (setting `isPlayed` before the calls) or maybe we could replace the icon by a spinner during the request?

Reply:
Updated